paimon-spark/paimon-spark3-common/src/main/scala/org/apache/spark/sql/paimon/shims/Spark3Shim.scala

```diff
@@ -16,57 +16,53 @@
  * limitations under the License.
  */
 
-package org.apache.spark.sql.paimon
+package org.apache.spark.sql.paimon.shims
+
+import org.apache.paimon.spark.data.{Spark3ArrayData, Spark3InternalRow, SparkArrayData, SparkInternalRow}
+import org.apache.paimon.types.{DataType, RowType}
 
 import org.apache.spark.sql.{Column, SparkSession}
-import org.apache.spark.sql.catalyst.{InternalRow => SparkInternalRow}
 import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
-import org.apache.spark.sql.catalyst.parser.{ParserInterface => SparkParserInterface}
-import org.apache.spark.sql.catalyst.plans.logical.{Aggregate => SparkAggregate}
-import org.apache.spark.sql.catalyst.util.{ArrayData => SparkArrayData}
-import org.apache.spark.sql.connector.catalog.{Identifier, Table, TableCatalog => SparkTableCatalog}
+import org.apache.spark.sql.catalyst.parser.ParserInterface
+import org.apache.spark.sql.catalyst.parser.extensions.PaimonSparkSqlExtensionsParser
+import org.apache.spark.sql.catalyst.plans.logical.Aggregate
+import org.apache.spark.sql.connector.catalog.{Identifier, Table, TableCatalog}
 import org.apache.spark.sql.connector.expressions.Transform
 import org.apache.spark.sql.types.StructType
 
 import java.util.{Map => JMap}
 
-/** Shims for Spark 3.x in [[org.apache.spark.sql]]. */
-object shims {
-
-  /** In [[org.apache.spark.sql.catalyst]]. */
+class Spark3Shim extends SparkShim {
 
-  abstract class ParserInterface extends SparkParserInterface {
-    val delegate: SparkParserInterface
+  override def createSparkParser(delegate: ParserInterface): ParserInterface = {
+    new PaimonSparkSqlExtensionsParser(delegate)
   }
 
-  abstract class ArrayData extends SparkArrayData {}
-
-  abstract class InternalRow extends SparkInternalRow {}
+  override def createSparkInternalRow(rowType: RowType): SparkInternalRow = {
+    new Spark3InternalRow(rowType)
```

Comment:
Maybe we can use a unified style, such as unifying `PaimonSpark3SqlExtensionsParser` `Spark3InternalRow`,
 or `PaimonSparkSqlExtensionsParser` `SparkInternalRow`.

Reply:
Right，I choose the first style.